tests/components/generic_hygrostat/test_config_flow.py

```diff
@@ -0,0 +1,106 @@
+"""Test the generic hygrostat config flow."""
+
+from unittest.mock import patch
+
+from syrupy.assertion import SnapshotAssertion
+from syrupy.filters import props
+
+from homeassistant.components.generic_hygrostat import (
+    CONF_DEVICE_CLASS,
+    CONF_DRY_TOLERANCE,
+    CONF_HUMIDIFIER,
+    CONF_NAME,
+    CONF_SENSOR,
+    CONF_WET_TOLERANCE,
+    DOMAIN,
+)
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+SNAPSHOT_FLOW_PROPS = props("type", "title", "result", "error")
+
+
+async def test_config_flow(hass: HomeAssistant, snapshot: SnapshotAssertion) -> None:
+    """Test the config flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result == snapshot(name="init", include=SNAPSHOT_FLOW_PROPS)
```

Comment:
Honestly, at this point this is just the same as

```suggestion
    assert result.type is FlowResultType.FORM
```

It's not that we assert huge data objects any more, so imo we should just not use snapshot here, but I won't block this

Reply:
It would be a few more checks like not none, title and stuff. Which is sort of what snapshot is for. Now the snapshot is quite readable. I would prefer to keep it like this for now.

I will take your objections to it into consideration though.